blobstore-uploader/src/main/java/org/jclouds/examples/blobstore/BlobUploader.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jclouds.examples.blobstore;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.Properties;
+import java.util.UUID;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FileUtils;
+import org.jclouds.Constants;
+import org.jclouds.ContextBuilder;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.domain.Blob;
+import org.jclouds.domain.Location;
+import org.jclouds.io.Payloads;
+import org.jclouds.openstack.swift.v1.blobstore.RegionScopedBlobStoreContext;
+
+public class BlobUploader implements Runnable {
+   /**
+    * ThreadLocal allows us to use 1 container and 1 connection per thread and reuse them.
+    * They need to be static, but here we also have to instantiate them at runtime, as we pass parameters from
+    * the command line.
+    */
+   private static ThreadLocal<BlobStore> blobStore = new ThreadLocal<BlobStore>();
+   private static ThreadLocal<String> container = new ThreadLocal<String>();

```

Comment:
To be reused by threadpool threads for multiple files. That way the same blobstore and container is used to upload multiple files.